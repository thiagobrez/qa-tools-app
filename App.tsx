import React from 'react';

import {
  View,
  Text,
  StyleSheet,
} from 'react-native';

import { Button } from '~/components';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'darkblue',
  },
  textWrapper: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 20,
  },
  title: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 30,
  },
  subtitle: {
    color: 'white',
    fontSize: 12,
  },
});

const App = () => (
  <View style={styles.container}>
    <View style={styles.textWrapper}>
      <Text style={styles.title}>
        qa-tools-app
      </Text>
      <Text style={styles.subtitle}>
        testing some tools
      </Text>
    </View>
    <Button round
            text="Click me"
    />
  </View>
);

export default App;
