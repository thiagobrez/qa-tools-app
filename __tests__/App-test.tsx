import 'react-native';
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

// eslint-disable-next-line import/no-unresolved
import App from '../App';

it('renders correctly', () => {
  render(<App />);
});

it('should click three times', async () => {
  const { getByTestId, getByText } = render(<App />);

  const button = getByTestId('click-me');
  const expectedValue = 'You clicked 3 times';

  fireEvent.press(button);
  fireEvent.press(button);
  fireEvent.press(button);

  expect(getByText(expectedValue)).toMatchSnapshot();
});
