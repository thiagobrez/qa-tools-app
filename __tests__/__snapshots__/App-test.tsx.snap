// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should click three times 1`] = `
<Text
  style={
    Object {
      "color": "white",
      "fontSize": 12,
    }
  }
>
  You clicked 
  3
   times
</Text>
`;
